src/LinAlg/matrix.h

```diff
@@ -971,6 +995,23 @@ namespace utl //! General utility classes and functions.
       return false;
     }
 
+    //! \brief Check dimension compatibility for matrix-matrix multiplication,
+    //! when the matrix A is represented by a one-dimensional vector.
+    bool compatible(const std::vector<T>& A, const matrix3d<T>& B,
+                    size_t& M, size_t& N, size_t& K)
+    {
+      N = B.n[1]*B.n[2];
+      K = B.n[0];
+      M = K > 0 ? A.size() / K : 0;
+      if (M*K == A.size() && !A.empty()) return true;
```

Comment:
reverse if

Reply:
not 100% sure, but I assume you here mean I shall check A.empty() first? Can change it.
Don't think it matters though, except avoiding one multiplication in the case A _is_ empty.
And btw, all the other compatible methods evaluates the if statement in this order, then they should be changed also, for consistency.